app/controllers/users_controller.rb

```diff
@@ -65,23 +63,21 @@ def update_language_settings
     end
   end
 
-  def destroy
-    set_user
+  def request_destroy
+    set_tabs("account")
+    Users::RequestDestroy.call(@user)
+    flash[:settings_notice] = "You have requested account deletion. Please, check your email for further instructions."
+    redirect_to "/settings/#{@tab}"
+  end
+
+  def confirm_destroy
+    destroy_token = RedisRailsCache.read("user-destroy-token-#{@user.id}")
+    raise ActionController::RoutingError, "Not Found" unless destroy_token.present? && destroy_token == params[:token]
+
```

Comment:
I think I am just missing something on how this all ties together but don't we want to then actually destroy the user here, after the confirmation? Where does the actual destroy after the confirm get kicked off?

Reply:
It's done in the `full_delete` action which runs after submitting a form (confirmation). Maybe I should add token to the form and include the check inside that action as well?
Btw I marked the pr as wip because I'm working on the tests and refinements.

Reply:
oh I get it, I thought this was the action doing the destroy, it's still one more step to get it done. 👍